server-core/service/src/main/java/io/infectnet/server/service/configuration/ServiceModule.java

```diff
@@ -0,0 +1,28 @@
+package io.infectnet.server.service.configuration;
+
+import dagger.Module;
+import dagger.Provides;
+import io.infectnet.server.persistence.configuration.PersistenceModule;
+import io.infectnet.server.service.TokenService;
+import io.infectnet.server.service.UserService;
+import io.infectnet.server.service.impl.TokenServiceImpl;
+import io.infectnet.server.service.impl.UserServiceImpl;
+
+import javax.inject.Singleton;
+
+@Module(includes = PersistenceModule.class)
+public class ServiceModule {
+
+    @Provides
+    @Singleton
+    TokenService providesTokenService() {
```

Comment:
Either look for a solution in the Dagger documentation (explicit public scope) or remove the rule from the codacy configuration.